mono/metadata/icall.c

```diff
@@ -9320,7 +9320,7 @@ ves_icall_System_GC_GetMaxGeneration (void)
 	return mono_gc_max_generation ();
 }
 
-gint64
+guint64
```

Comment:
it's declared as `long` on the managed side
https://github.com/mono/mono/blob/8fed6666b080d3d5634fd9068cd42f273f819eeb/mcs/class/referencesource/mscorlib/system/gc.cs#L108
and we can't change the signature https://docs.microsoft.com/en-us/dotnet/api/system.gc.getallocatedbytesforcurrentthread?view=netframework-4.8#System_GC_GetAllocatedBytesForCurrentThread

Reply:
Does changing it to gint64 actually fix anything? You still end up with a weird negative value sometimes, its just a matter of when.

Reply:
If 63 bits isn't enough, then 64 probably isn't either?, is kinda my saying, usually with 31 and 32, widen to 63 or 64..

Reply:
Managed tends to prefer signed, because I guess, to avoid too many confusing integer types, and I not sure I sympathetic to that or not.
(similar to array/string lengths not being 64 bits, alas).

Reply:
And then if managed is "wrong", should native go down with the ship, or correct where possible?
And is improvement possible? I mean, the main consumer of the value will be the managed side anyway?

When the gint64 overflows, the result is undefined.
Some implementations might trap. Anything mono runs on? Anything real?
When guint64 overflows, the result is well defined, silent modulo math.
Not sure it matters.
Will the 63 bits overflow in our lifetime?

Reply:
I still wonder if mono's predilection for signed numbers is wrong or harmful.
Google style guide is similar though, prefers signed 32bit `int` above all other integer types.

Reply:
63 bits is already overflowing in a test suite, which is where this PR comes from. I guess if other tests run and do lots of allocation before hand.

Reply:
How long before 64 bits overflows?
Is something larger needed?

Reply:
@naricc are you sure the number is overflowing and there isn't any bug in the implementation. What suite is overflowing ?

Reply:
Well, any way, I don't think this PR actually can fix the issue. So I am abandoning it.